posit-dev/positron
# Track reasons for R discovery and rejection and do stuff with them

Addresses #1397 
Addresses #5220 
Continued preparation for #5223

A lot of the changes are very internal, but there are also user-facing improvements:

* Even more effort to track the reasons (plural) why we've discovered an R installation and, perhaps, the reason an R installation can't be used with Positron. The richer information appears in the Positron R Extension output channel.
* If we discover at least one R installations, but there is no usable R installation, we put a pop-up warning. The exact details of this will be good to discuss in the next R meeting.

### QA Notes

To experience this, you need to have at least 1 unusable R installation around and 0 usable ones.

For rig-installed R, use `rig rm release` (or similar) to remove an R installation. Repeat as necessary, until all usable R is gone!

Older R installers are here for macOS arm64: https://cran.rstudio.com/bin/macosx/big-sur-arm64/base/. This makes it easy to get R 4.1, for example, which is below our minimum supported version.

You should see a modal telling you that all R installations are unusable, allowing you to look at the Positron R Extension output channel or dismiss.
